vertexai/src/LiveGenerativeModel.cs

```diff
@@ -0,0 +1,131 @@
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.MiniJSON;
+
+namespace Firebase.VertexAI {
+
+/// <summary>
+/// A live, generative AI model for real-time interaction.
+///
+/// See the [Cloud
+/// documentation](https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/multimodal-live)
+/// for more details about the low-latency, two-way interactions that use text,
+/// audio, and video input, with audio and text output.
+///
+/// > Warning: For Vertex AI in Firebase, Live Model
+/// is in Public Preview, which means that the feature is not subject to any SLA
+/// or deprecation policy and could change in backwards-incompatible ways.
+/// </summary>
+public class LiveGenerativeModel {
+  private readonly FirebaseApp _firebaseApp;
+
+  // Various setting fields provided by the user.
+  private readonly string _location;
+  private readonly string _modelName;
+  private readonly LiveGenerationConfig? _liveConfig;
+  private readonly Tool[] _tools;
+  private readonly ModelContent? _systemInstruction;
+  private readonly RequestOptions? _requestOptions;
+
+  /// <summary>
+  /// Intended for internal use only.
+  /// Use `VertexAI.GetLiveModel` instead to ensure proper initialization and configuration of the `LiveGenerativeModel`.
+  /// </summary>
+  internal LiveGenerativeModel(FirebaseApp firebaseApp,
+                               string location,
+                               string modelName,
+                               LiveGenerationConfig? liveConfig = null,
+                               Tool[] tools = null,
+                               ModelContent? systemInstruction = null,
+                               RequestOptions? requestOptions = null) {
+    _firebaseApp = firebaseApp;
+    _location = location;
+    _modelName = modelName;
+    _liveConfig = liveConfig;
+    _tools = tools;
+    _systemInstruction = systemInstruction;
+    _requestOptions = requestOptions;
+  }
+
+  /// <summary>
+  /// Establishes a connection to a live generation service.
+  ///
+  /// This function handles the WebSocket connection setup and returns an `LiveSession`
+  /// object that can be used to communicate with the service.
+  /// </summary>
+  /// <param name="cancellationToken">The token that can be used to cancel the creation of the session.</param>
+  /// <returns>The LiveSession, once it is established.</returns>
+  public async Task<LiveSession> ConnectAsync(CancellationToken cancellationToken = default) {
+    ClientWebSocket clientWebSocket = new();
+
+    string endpoint = $"wss://firebasevertexai.googleapis.com/ws/google.firebase.vertexai.v1beta.LlmBidiService/BidiGenerateContent/locations/{_location}?key={_firebaseApp.Options.ApiKey}";
+
+    // Set initial headers
+    clientWebSocket.Options.SetRequestHeader("x-goog-api-client", "genai-csharp/0.1.0");
```

Comment:
'0.1.0' be runtime version?

Reply:
I copied the TODO I have from GenerativeModel. Basically, need to expose some info from C++ for this.